packages/core/src/content-serdes.ts

```diff
@@ -162,16 +171,16 @@ export class ContentSerdes {
         const par = ContentSerdes.getMediaTypeParameters(contentType);
 
         // choose codec based on mediaType
-        if (this.codecs.has(mt)) {
+        const codec = this.codecs.get(mt);
+        if (codec) {
             console.debug("[core/content-serdes]", `ContentSerdes serializing to ${contentType}`);
-            const codec = this.codecs.get(mt);
             bytes = codec.valueToBytes(value, schema, par);
         } else {
             console.warn(
                 "[core/content-serdes]",
                 `ContentSerdes passthrough due to unsupported serialization format '${contentType}'`
             );
-            bytes = Buffer.from(value.toString());
+            bytes = Buffer.from(value === null ? "" : value.toString());
```

Comment:
Is it intended to only check for `null` and not also for `undefined` (using `value == null`) here?

Reply:
So if we just trust the tsc compiler we know that value is not undefined, therefore it does not make sense. Some people say that in general is not guaranteed that people will use your library from TS and therefore you should take into account corner cases that are covered by the compiler. My problem here is that we don't have a policy. So if we check for undefined here, then we have to do it everywhere basically.... not sure if is something that we want.

Reply:
Hmm, there is a check above, though, which logs a warning if the value is `undefined`. To be consistent, that check would need to be removed as well?

My suggestion here would be to simply use the following:

```suggestion
            bytes = Buffer.from(value?.toString() ?? "");
```

This way, the code is a bit more concise and `undefined` is also covered (which would otherwise cause an error, as `toString()` seems to not exist on `undefined`, in contrast to other data types).

Reply:
I might have agreed with @JKRhb's suggestion but then I thought more about the logic. IMHO is wrong... The else branch is activated if no codec is found and it defaults to simple sending value.toString or "". But value.toString() is not always correct. For example for objects it returns: [object Object] . Why should we send this to the wire? 🤔 So the resolve this conversation I would like to open an issue instead of fixing this detail. thumbs up if you agree